kernel/lib/printf.cpp

```diff
@@ -288,6 +294,10 @@ OK, I found my mistake. The math here is _always_ unsigned */
             default:
                 break;
             }
+            // Tell the compiler that this isn't an issue
+            // (at least I'm assuming it isn't)
+            [[fallthrough]];
```

Comment:
Maybe explain to the person looking at the code why this is the case as well

Reply:
I'm not certain but I think it's because we want the code below to execute if it falls through so that we continue to print. If you have a better guess / comment just leave it here and I'll add it.

Reply:
Ah okay yeah. It's kinda a horribly structured function in the first place.

Reply:
Yeah, it is, but I don't feel like optimizing it right now.

Reply:
All good